### PR TITLE
dns: packed dns_header_s and dns_question_s

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

avoid the unalignment issue

## Impact

Minor

## Testing

Pass CI

### CODE REVIEW

File: include/nuttx/net/dns.h [hunk 1]
```diff
@@ -147,15 +147,15 @@ struct dns_header_s
   uint16_t numanswers;
   uint16_t numauthrr;
   uint16_t numextrarr;
-};
+}end_packed_struct;
```
Comment (pkarashchenko) [inline include/nuttx/net/dns.h:150]:
```suggestion
} end_packed_struct;
```

File: include/nuttx/net/dns.h [hunk 2]
```diff
@@ -147,15 +147,15 @@ struct dns_header_s
...
 
 /* The DNS question message structure */
 
-struct dns_question_s
+begin_packed_struct struct dns_question_s
 {
   uint16_t type;
   uint16_t class;
-};
+}end_packed_struct;
```
Comment (pkarashchenko) [inline include/nuttx/net/dns.h:158]:
```suggestion
} end_packed_struct;
```

File: include/nuttx/net/dns.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/nuttx/net/dns.h]: Done.

File: include/nuttx/net/dns.h (hunk 2 above)
Comment (xiaoxiang781216) [reply include/nuttx/net/dns.h]: Done.

File: libs/libc/netdb/lib_dnsquery.c [hunk 3]
```diff
@@ -438,13 +437,12 @@ static int dns_recv_response(int sd, FAR union dns_addr_u *addr, int naddr,
   /* Validate query type and class */
 
   que = (FAR struct dns_question_s *)nameptr;
-  memcpy(&bak, que, sizeof(struct dns_question_s));
 
   /* N.B. Unaligned access may occur here */
```
Comment (pkarashchenko) [inline libs/libc/netdb/lib_dnsquery.c:441]: Should we still keep this comment here?

**Review by pkarashchenko — APPROVED** [review]

File: libs/libc/netdb/lib_dnsquery.c (hunk 3 above)
Comment (xiaoxiang781216) [reply libs/libc/netdb/lib_dnsquery.c:441]: Yes, since que may be unaligned.

**Review by anchao — APPROVED** [review]

**Pull request merged**